unit_test/entities/controller/test_controller_class.py

```diff
@@ -40,12 +40,14 @@ def setUp(self):
         self.producer1.priority = 1
         self.producer1.temperature_return = 20.0
         self.producer1.temperature_supply = 30.0
+        self.producer1.marginal_costs = 0.5
         self.producer2 = Mock()
         self.producer2.id = "producer2"
         self.producer2.power = 4.0
         self.producer2.priority = 2
         self.producer2.temperature_return = 40.0
         self.producer2.temperature_supply = 50.0
+        self.producer2.marginal_costs = 0.8
```

Comment:
self.controller = NetworkController(
            [self.producer1, self.producer2], [self.consumer1, self.consumer2], [self.storage1]
        )
        add the variable names to make it clear what you are doing.

Reply:
done